tools/Microsoft.AspNetCore.Server.Kestrel.GeneratedCode/KnownHeaders.cs

```diff
@@ -31,19 +31,27 @@ static string If(bool condition, Func<string> formatter)
                     case {byLength.Key}:
                         {{{Each(byLength, header => $@"
                             if ({header.EqualIgnoreCaseBytes()})
-                            {{
+                            {{{(header.Identifier == "ContentLength" ? $@"
+                                if (ContentLength.HasValue)
+                                {{
+                                    ThrowInvalid{(className == "FrameResponseHeaders" ? "Response" : "Request")}ContentLengthException(AppendValue(HeaderUtilities.FormatInt64(ContentLength.Value), value).ToString());
```

Comment:
The logging is broken when throwing a BadHttpRequestException without using RejectRequest() for cases when the exception can't easily be thrown from inside the Frame class as is the case here.

A situation very similar to the one you're running into now can be found [elsewhere in this class](https://github.com/aspnet/KestrelHttpServer/blob/10fe5e6fa25e95ab9acd2b3bd733e78be4315a37/tools/Microsoft.AspNetCore.Server.Kestrel.GeneratedCode/KnownHeaders.cs#L58).

[This commit](https://github.com/aspnet/KestrelHttpServer/commit/78584799a412ef24327dca5b3c62315d708bb44c) regressed the logging, but I can't find the PR that changed it. The commit seems to have been pushed sometime between #1163 and #1166.

@cesarbs Can you take a look? We can create an issue if it requires a big change.